### PR TITLE
Feature: Implemented Notification Settings

### PR DESCRIPTION
**Pull request by NecyNazy**
### **How should this be manually tested?**
Try accessing a protected route
* Without Being Pre-Authorized
* With Invalid email

The expected result is to get a 401 Unauthorized response

### **Checklist of What You Did**

 - [ ] Define and implement the database schema for notification settings.
 - [ ] Develop and test database migrations scripts to apply the new schema.
 - [ ] Implement secure and validated API endpoints for CRUD operations on notification settings.
 - [ ] Add  authentication to the POST and GET endpoints for notification settings.
 - [ ] Write Unit and E2E Tests for notification settings service  

### **Reference Issue**
[[FEAT]: Implement Backend for Notification Settings](https://github.com/hngprojects/hng_boilerplate_java_web/issues/20)

### CODE REVIEW

File: src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping
+public class NotificationSettingsController {
+    @Autowired
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java:18]: Use constructor injection

File: src/main/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import hng_java_boilerplate.notificationSettings.entity.NotificationSettings;
+import hng_java_boilerplate.notificationSettings.repository.NotificationSettingsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class NotificationSettingsService {
+    @Autowired
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsService.java:12]: Use constructor injection

File: src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+import java.util.Optional;
+import java.util.UUID;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class NotificationSettingsServiceTest {
```
Comment (Fifetoyi) [inline src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java:23]: Include e2e tests

File: pom.xml
Comment (Fifetoyi) [inline pom.xml:1]: Too many changes in the pom file. Revert back and only add the dependencies you need

File: src/main/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsService.java (hunk 2 above)
Comment (NecyNazy) [reply src/main/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsService.java]: done

File: src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java (hunk 1 above)
Comment (NecyNazy) [reply src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java:18]: done

File: src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java (hunk 3 above)
Comment (NecyNazy) [reply src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java:23]: done

File: pom.xml
Comment (NecyNazy) [reply pom.xml:1]: done

File: src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java (hunk 1 above)
Comment (Fifetoyi) [reply src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java:18]: I can't see constructor injection. Use @RequiredArgsConstructor and make the field final

File: src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java (hunk 3 above)
Comment (Fifetoyi) [reply src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java:23]: Still only unit tests here

Comment (NecyNazy) [reply src/test/java/hng_java_boilerplate/notificationSettings/service/NotificationSettingsServiceTest.java:23]: what do i do, ive made the changes and pushed... i didnt see any prompt to fetch changes. what do i do?

File: src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java (hunk 1 above)
Comment (NecyNazy) [reply src/main/java/hng_java_boilerplate/notificationSettings/controller/NotificationSettingsController.java]: updated with @requiredargsconstructor

**Comment by Am0du** [comment]
Follow buka's guideline

**Comment by NecyNazy** [comment]
> Follow buka's guideline

I have followed the guideline. Kindly revert.

**Pull request closed without merging**